PWGCF/FemtoDream/femtoDreamProducerReducedTask.cxx

```diff
@@ -133,16 +135,33 @@ struct femtoDreamProducerReducedTask {
   float getMagneticFieldTesla(uint64_t timestamp)
   {
     // TODO done only once (and not per run). Will be replaced by CCDBConfigurable
-    static o2::parameters::GRPObject* grpo = nullptr;
-    if (grpo == nullptr) {
-      grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>("GLO/GRP/GRP", timestamp);
+    float output = -999;
+
+    if (ConfIsRun3 && !ConfIsMC) {
+      static o2::parameters::GRPMagField* grpo = nullptr;
+      if (grpo == nullptr) {
+        grpo = ccdb->getForTimeStamp<o2::parameters::GRPMagField>("GLO/Config/GRPMagField", timestamp);
+        if (grpo == nullptr) {
+          LOGF(fatal, "GRP object not found for timestamp %llu", timestamp);
+          return 0;
+        }
+        LOGF(info, "Retrieved GRP for timestamp %llu with magnetic field of %d kG", timestamp, grpo->getL3Current());
+      }
+      output = 0.1 * (grpo->getL3Current());
```

Comment:
You need to consider that most probably you are getting the L3 current and not the magnetic field
If I'm correct, the new layout still does not consider extracting the actual magnetic field valu